### ISSUE TITLE
incorrect array into object destructuring

### ISSUE THREAD

**Issue by a-x-** [eslint/eslint#10505]
**what the rule should do:**
Warn or disallow well-known array into object destructuring cases

**What category of rule is this?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**2-3 code examples that this rule will warn about:**

```js
(...params) => ({  ...params })
```

```js
const arr = [];
{ ...arr }
```

**Why should this rule be included in ESLint (instead of a plugin)?**

afaik, it looks as basic simple check

**Comment by a-x-** [comment]
It might be there is covering rule/plugin I don't know, I'm sorry about it.
I'll be happy to know

**Comment by ilyavolodin** [comment]
I think this rule would require type checking, which ESLint currently doesn't support. So I'm not sure we can create a rule like this.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
